src/numeric_index.c

```diff
@@ -471,38 +462,28 @@ int NumericRangeNode_RemoveChild(NumericRangeNode **node, NRN_AddRv *rv) {
     return CHILD_NOT_EMPTY;
   }
 
-  rv->changed = 1;
-
-  // we can remove local and use child's instead
-  if (n->range) {
-    if (n->range->entries->numDocs != 0) {
-      return CHILD_NOT_EMPTY;
-    }
-    removeRange(n, rv);
+  if (n->range && n->range->entries->numDocs != 0) {
+    // We are on a non-leaf node, with some data in it but some of its children are empty.
+    // Ideally we would like to trim the empty children, but today we don't fix missing ranges
+    // of inner nodes, so we better keep the node as is.
+    // TODO: remove this block when we fix the missing ranges issue.
+    return CHILD_NOT_EMPTY;
   }
 
-  // both children are empty, save one as parent
-  if (rvRight == CHILD_EMPTY && rvLeft == CHILD_EMPTY) {
-    rm_free(n);
-    *node = rightChild;
-    NumericRangeNode_Free(leftChild, rv);
-
-    return CHILD_EMPTY;
-  }
+  rv->changed = 1;
 
-  // one child is not empty, save copy as parent and free
+  // at least one child is empty. keep an empty child and replace the parent with the other child
   if (rvRight == CHILD_EMPTY) {
-    // right child is empty, save left as parent
-    rm_free(n);
-    *node = leftChild;
-    NumericRangeNode_Free(rightChild, rv);
+    // right child is empty, save left as parent (might be empty)
+    *node = n->left;
+    n->left = NULL; // avoid freeing it
   } else {
     // left child is empty, save right as parent
-    rm_free(n);
```

Comment:
where did the rm_free(n) get moved to?

Reply:
nvm, found it

Reply:
this `rm_free` call, along with the previous call to `removeRange`, was moved to the last call to `NumericRangeNode_Free` which does exactly that